src/main/java/frc/robot/Constants.java

```diff
@@ -159,9 +161,51 @@
         public static int arbAnalogPerRadian = 10;
     }
 
+    public static final class SolverPID{
+
+        public static int kPIDLoopIdx = 0; // The loop Index
+        public static int kTimeoutMs = 30; // The timeout to wait when writing variables to the motors
+        public static boolean sensorPhase = true; // The phase of the sensor
+        public static boolean motorInvert = false;
+        public static final Gains kGains = new Gains(0.15,  // kP TODO
+                                                     0.0,   // kI
+                                                     1.0,   // kD
+                                                     0.0,   // kF
+                                                     0,     // kIzone
+                                                     1.0);  // kPeakoutput
+        // Measurements and other
+        public static int TicksPerRotation = 10000;        // Calibrated 12/16/2019
+    }
+
+    public static final class Colors{
+        public static final class Blue{
+            public static double red = 0.0;
+            public static double green = 0.0;
+            public static double blue = 0.0;
+        }
+
+        public static final class Yellow{
+            public static double red = 0.0;
+            public static double green = 0.0;
+            public static double blue = 0.0;
+        }
+
+        public static final class Red{
+            public static double red = 0.0;
+            public static double green = 0.0;
+            public static double blue = 0.0;
+        }
+
+        public static final class Green{
+            public static double red = 0.0;
+            public static double green = 0.0;
+            public static double blue = 0.0;
+        }
+    }
+
```

Comment:
For my understanding @KenwoodFox ~~

How come a lot of this is nested? 

This part of the program looks fantastic other than some tuning!

Reply:
Yellow, Red and Blue are subclasses of the class color
this allows you to say Colors.Blue, Colors.Red and Colors.Green, allowing better readablity